app/components/icon/icon.props.ts

```diff
@@ -17,5 +17,5 @@ export interface IconProps {
    * The name of the icon
    */
 
-  icon?: IconTypes
+  icon: IconTypes
```

Comment:
I think we can force the user of `<Icon>` to always provide the icon type. This saves one null check.